src/main/java/net/datafaker/idnumbers/NorwegianIdNumber.java

```diff
@@ -0,0 +1,126 @@
+import static net.datafaker.idnumbers.Utils.randomGender;
+
+/**
+ * The Norwegian Individual ID Number is 11 digits log
+ * Format: ddMMyyCCGkk
+ * <ul>
+ *     <li>dd - day of birth</li>
+ *     <li>MM - month of birth</li>
+ *     <li>yy - year of birth</li>
+ *     <li>CC - sequence (range depends on century)</li>
+ *     <li>kk - checksum</li>
+ * </ul>
+ * <p>
+ *     Example: 11077941012
+ * </p>
+ * @see <a href="https://www.skatteetaten.no/en/person/national-registry/identitetsnummer-og-elektronisk-id/fodselsnummer/">Overview</a>
+ * @see <a href="https://taxid.pro/">Online generator</a>
+ * @see <a href="https://www.oecd.org/content/dam/oecd/en/topics/policy-issue-focus/aeoi/norway-tin.pdf">More on centuries</a>
+ */
+public class NorwegianIdNumber implements IdNumberGenerator {
+    private static final DateTimeFormatter BIRTHDAY_FORMAT = DateTimeFormatter.ofPattern("ddMMyy");
+    private static final List<Map.Entry<Interval, Interval>> CENTURIES = Stream.of(
+            "1854-1899, 74-50",
+            "1940-1999, 99-90",
+            "1900-1999, 49-0",
+            "2000-2039, 99-50"
+        )
+        .map(s -> s.split(", "))
+        .map(array -> Map.entry(Interval.of(array[0]), Interval.of(array[1]).reverse()))
+        .toList();
+
+    private static final int[] CHECKSUM_COEFFICIENTS_K1 = {3, 7, 6, 1, 8, 9, 4, 5, 2};
+    private static final int[] CHECKSUM_COEFFICIENTS_K2 = {5, 4, 3, 2, 7, 6, 5, 4, 3, 2};
+
+    @Override
+    public String countryCode() {
+        return "NO";
+    }
+
+    @Override
+    public PersonIdNumber generateValid(BaseProviders faker, IdNumberRequest request) {
+        LocalDate birthday = birthday(faker, request);
+        Gender gender = gender(faker, request);
+
+        String basePart = basePart(faker, birthday, gender);
+        String idNumber = "%s%02d".formatted(basePart, checksum(basePart));
+        return new PersonIdNumber(idNumber, birthday, randomGender(faker));
+    }
+
+    @Override
+    public String generateInvalid(BaseProviders faker) {
+        String valid = generateValid(faker);
+        String basePart = valid.substring(0, valid.length() - 2);
+        int invalidChecksum = (checksum(basePart) + 1) % 100;
+        return "%s%02d".formatted(basePart, invalidChecksum);
+    }
+
+    String basePart(BaseProviders faker, LocalDate birthday, Gender gender) {
+        String birthdayDigits = BIRTHDAY_FORMAT.format(birthday);
+
+        int sequenceNumber = CENTURIES
+            .stream()
+            .filter(e -> e.getKey().contains(birthday.getYear()))
+            .map(Map.Entry::getValue)
+            .map(interval -> faker.random().nextInt(interval.from, interval.to))
+            .findFirst()
+            .orElseThrow(() -> new IllegalArgumentException("Birthday is not in range of supported in Norway"));
+
+        int genderDigit = gender == Gender.FEMALE
+            ? faker.options().option(0, 2, 4, 6, 8)
+            : faker.options().option(1, 3, 5, 7, 9);
```

Comment:
```suggestion
        int genderDigit = faker.options().option(0, 2, 4, 6, 8);
        genderDigit = gender == Gender.FEMALE ? genderDigit: genderDigit + 1;
```

Reply:
Also possible. But IMHO this way is a bit harder to read.